src/plugins/data/public/data_sources/datasource_selector/datasource_selectable.tsx

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React, { useEffect } from 'react';
+import { DataSourceSelector } from './datasource_selector';
+import { DataSourceType } from '../datasource_services';
+import { DataSourceGroup, DataSourceOption, DataSourceSelectableProps } from './types';
+import { ISourceDataSet } from '../datasource/types';
+import { IndexPattern } from '../../index_patterns';
+
+// Mapping between datasource type and its display name.
+const DATASOURCE_TYPE_DISPLAY_NAME_MAP = {
+  DEFAULT_INDEX_PATTERNS: 'Index patterns',
+  s3glue: 'Amazon S3',
+  spark: 'Spark',
+};
+
+type DataSetType = string | IndexPattern;
+
+export const DataSourceSelectable = ({
+  dataSources, // list of all available datasource connections.
+  dataSourceOptionList, // combo box renderable option list derived from dataSources
+  selectedSources, // current selected datasource in the form of [{ label: xxx, value: xxx }]
+  setSelectedSources,
```

Comment:
nit: can we call this prop `onDataSourceSelect`? This is a callback prop, so the `on` prefix makes more sense here